sympy/core/numbers.py

```diff
@@ -1691,6 +1691,49 @@ def limit_denominator(self, max_denominator=1000000):
         f = fractions.Fraction(self.p, self.q)
         return Rational(f.limit_denominator(fractions.Fraction(int(max_denominator))))
 
+    def limit_num_denom(self, a, b = None):
+        """Closest Rational to self with numerator at most a
+           and denominator at most b if not None else Closest
+           Rational to self with denominator at most a.
+
+        Examples
+        ========
+
+        >>> from sympy import Rational
+        >>> Rational(1819831678027117, 9007199254740992).limit_num_denom(10**5)
+        11003/54459
+        >>> Rational(1819831678027117, 9007199254740992).limit_num_denom(10**3, 10**5)
+        376/1861
+
+        """
+        if a < 0:
+            raise AssertionError("Value should be greater than 0")
+
+        if self == 0:
+            return self
+        elif self < 0:
```

Comment:
I was especially thinking about testing this line, which I do not think happens at the moment.